yazi-config/src/theme/filetype.rs

```diff
@@ -9,13 +9,33 @@ use crate::Pattern;
 pub struct Filetype {
 	pub name:  Option<Pattern>,
 	pub mime:  Option<Pattern>,
+	pub typ:   Option<String>,
 	pub style: Style,
 }
 
 impl Filetype {
 	pub fn matches(&self, path: &Path, mime: Option<&str>) -> bool {
 		let is_dir = mime == Some(MIME_DIR);
-		self.name.as_ref().is_some_and(|n| n.match_path(path, is_dir))
+		let special_types = self.typ.as_ref().is_some_and(|t| match t.as_str() {
+			"symlink" => path.is_symlink(),
+			#[cfg(unix)]
+			"executable" => {
+				use std::os::unix::fs::PermissionsExt;
+				let Ok(metadata) = path.metadata() else {
+					return false;
+				};
+
+				let mode_bin = format!("{:b}", metadata.permissions().mode());
+				return mode_bin.len() == 16 && match *mode_bin.as_bytes() {
+					[.., _, _, a, _, _, b, _, _, c] => a == b'1' || b == b'1' || c == b'1',
```

Comment:
This is the most intuitive way I got it to work...

Reply:
i think you can use smth like
``` rust
metadata.permissions().mode() & 0o111 != 0
```
or `0o100` just for user-executable bit

Reply:
Right, I didn't think about that... Thank you for the advice!